File: 프로그래머스/1/140108. 문자열 나누기/문자열 나누기.py
from collections import deque

def solution(s):
    my_queue = deque(s)
    result = 0
    
    while my_queue:
        first_str = my_queue[0]
        first_str_count = 1
        another_str_count = 0
        
        for i in range(1, len(my_queue)):
            if first_str == my_queue[i]:
                first_str_count += 1
            else:
                another_str_count += 1

            if first_str_count == another_str_count:
                for j in range(first_str_count * 2):
                    my_queue.popleft()
                result += 1
                break
        else:
            result += 1
            break
            
    return result
            
        
        
                                
        
        
    
            
        
        
        
        
    
    

File: 프로그래머스/1/140108. 문자열 나누기/문자열 나누기.js
function solution(s) {
    let result = 0;
    let cur_str_num = 0;
    let ant_str_num = 0;
    let cur_str = s[0]; 
    
    for(let i=0; i<s.length; i++) {
        if (cur_str !== s[i]) { 
            ant_str_num += 1; 
        } else {
            cur_str_num += 1;
        }
        
        if (cur_str_num === ant_str_num) {
            result += 1;
            cur_str_num = 0;
            ant_str_num = 0;
            
            if (i + 1 < s.length) {
                cur_str = s[i+1];
            }
        }
    }
    if (cur_str_num) result ++;
    return result;
}